protocols/gossipsub/src/behaviour.rs

```diff
@@ -1521,6 +1521,32 @@ where
         tracing::debug!(peer=%peer_id, "Completed GRAFT handling for peer");
     }
 
+    fn handle_extensions(&mut self, peer_id: &PeerId, extensions: Extensions) {
+        let Some(peer) = self.connected_peers.get_mut(peer_id) else {
+            tracing::error!(
+                peer=%peer_id,
+                "Extensions by unknown peer"
+            );
+            return;
+        };
+
+        if peer.extensions.is_some() {
+            tracing::debug!(
+                peer=%peer_id,
+                "Peer had already sent us extensions message"
+            );
+            return;
+        }
+
+        peer.extensions = Some(extensions);
+
+        if extensions.test_extension.unwrap_or(false)
+            && matches!(peer.kind, PeerKind::Gossipsubv1_3)
+        {
+            self.send_message(*peer_id, RpcOut::TestExtension);
```

Comment:
I understand the need for the `TestExtension`. However I dont think we should be sending this thing around to all peers on prod. 

I'd suggest hiding it behind a feature-gate or a configuration so that we don't send any extensions by default, but if we need to test something we can enable the feature. I.e I dont think the default state should be sending a TestExtension. 

One we have actual extensions, we should hopefully be able to remove this extension entirely.